packages/utils/src/post-process.ts

```diff
@@ -1,40 +1,382 @@
+      const [replacementQuasis, replacementExpressions] = node.quasis.reduce<
+        [TemplateElement[], Expression[]]
+      >(
+        ([elements, expressions], quasi, index) => {
+          // Note: Template literals have two variants, "cooked" and "raw". Here
+          // we use the cooked version.
+          // https://exploringjs.com/impatient-js/ch_template-literals.html#template-strings-cooked-vs-raw
+          const tokens = breakTokensTemplateLiteral(
+            quasi.value.cooked as string,
+          );
+
+          // Only update the node if something changed.
+          if (tokens[0].length <= 1) {
+            return [
+              [...elements, quasi as TemplateElement],
+              [...expressions, node.expressions[index] as Expression],
+            ];
+          }
+
+          return [
+            [...elements, ...(tokens[0] as TemplateElement[])],
+            [
+              ...expressions,
+              ...tokens[1],
+              node.expressions[index] as Expression,
+            ],
+          ];
+        },
+        [[], []],
+      );
+
+      path.replaceWith(
+        templateLiteral(
+          replacementQuasis,
+          replacementExpressions.filter((e) => e !== undefined),
+        ) as Node,
+      );
+
+      path.setData('visited', true);
+    },
+
+    StringLiteral(path) {
+      const { node } = path;
+
+      // Break up tokens that could be parsed as HTML comment terminators, or
+      // `import()` statements.
+      // For reference:
+      // - https://github.com/endojs/endo/blob/70cc86eb400655e922413b99c38818d7b2e79da0/packages/ses/error-codes/SES_HTML_COMMENT_REJECTED.md
+      // - https://github.com/MetaMask/snaps-skunkworks/issues/505
+      const tokens = breakTokens(node.value);
+
+      // Only update the node if the string literal was broken up.
+      if (tokens.length <= 1) {
+        return;
+      }
+
+      const replacement = tokens
+        .slice(1)
+        .reduce<Expression>(
+          (acc, value) => binaryExpression('+', acc, stringLiteral(value)),
+          stringLiteral(tokens[0]),
+        );
+
+      path.replaceWith(replacement as Node);
+      path.skip();
+    },
+
+    BinaryExpression(path) {
+      const source = path.getSource();
+
+      // Throw an error if HTML comments are used as a binary expression.
+      if (source.includes('<!--') || source.includes('-->')) {
+        throw new Error(
+          'Using HTML comments (`<!--` and `-->`) as operators is not allowed. The behaviour of ' +
+            'these comments is ambiguous, and differs per browser and environment. If you want ' +
+            'to use them as operators, break them up into separate characters, i.e., `a-- > b` ' +
+            'and `a < ! --b`.',
+        );
+      }
+    },
+  };
+
+  try {
+    const file = transformSync(code, {
+      // Prevent Babel from searching for a config file.
+      configFile: false,
+      comments: !stripComments,
```

Comment:
Do we not want `parserOpts: { attachComment: !stripComments }`? We don't need to parse them if we are throwing them out right?

Reply:
There's no `attachComment` in `parserOpts` 😕